components/brave_news/browser/brave_news_engine.cc

```diff
@@ -0,0 +1,261 @@
+      history_querier_(std::move(history_querier)) {
+  DETACH_FROM_SEQUENCE(sequence_checker_);
+  sequence_checker_.EnableStackLogging();
+}
+
+BraveNewsEngine::~BraveNewsEngine() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+}
+
+void BraveNewsEngine::GetLocale(SubscriptionsSnapshot snapshot,
+                                m::GetLocaleCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetPublishersController()->GetLocale(snapshot, std::move(callback));
+}
+
+void BraveNewsEngine::GetSignals(SubscriptionsSnapshot snapshot,
+                                 m::GetSignalsCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  auto* builder = MaybeFeedV2Builder();
+  CHECK(builder);
+
+  builder->GetSignals(snapshot, std::move(callback));
+}
+
+void BraveNewsEngine::GetPublishers(SubscriptionsSnapshot snapshot,
+                                    m::GetPublishersCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetPublishersController()->GetOrFetchPublishers(snapshot,
+                                                  std::move(callback));
+}
+
+void BraveNewsEngine::GetPublisherForSite(SubscriptionsSnapshot snapshot,
+                                          GURL site_url,
+                                          GetPublisherCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetPublishersController()->GetPublisherForSite(snapshot, site_url,
+                                                 std::move(callback));
+}
+
+void BraveNewsEngine::GetPublisherForFeed(SubscriptionsSnapshot snapshot,
+                                          GURL feed_url,
+                                          GetPublisherCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetPublishersController()->GetPublisherForFeed(snapshot, feed_url,
+                                                 std::move(callback));
+}
+
+void BraveNewsEngine::EnsurePublishersIsUpdating(
+    SubscriptionsSnapshot snapshot) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetPublishersController()->EnsurePublishersIsUpdating(snapshot);
+}
+
+void BraveNewsEngine::GetChannels(SubscriptionsSnapshot snapshot,
+                                  m::GetChannelsCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetChannelsController()->GetAllChannels(snapshot, std::move(callback));
+}
+
+void BraveNewsEngine::GetSuggestedPublisherIds(
+    SubscriptionsSnapshot snapshot,
+    m::GetSuggestedPublisherIdsCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  GetSuggestionsController()->GetSuggestedPublisherIds(snapshot,
+                                                       std::move(callback));
+}
+
+void BraveNewsEngine::GetFeed(SubscriptionsSnapshot snapshot,
+                              m::GetFeedCallback callback) {
+  if (auto* builder = MaybeFeedV1Builder()) {
```

Comment:
Does this need `DCHECK_CALLED_ON_VALID_SEQUENCE` too? Or deliberately skipped for FeedV1?

Reply:
Ah, just left it off because it wasn't directly accessing one of the guarded_by_sequence variable - added to be more explicit!